### ISSUE TITLE
How to limit YOLO.val to specific classes?

### ISSUE THREAD

**Issue by caisamuels** [ultralytics/ultralytics#20249]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I am trying to use YOLO.val() to evaluate the performance of the YOLOv11 model, but the model detects objects I didn't label in the ground truth. When the evaluation runs, these extra detections are counted as False Positives because they don't match any ground truth label, which lowers the overall Precision and F1 score.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @caisamuels, thank you for reaching out and using Ultralytics 🚀! This is an automated response to help you get started. An Ultralytics engineer will also review your question and provide direct assistance soon.

We recommend checking the [Docs](https://docs.ultralytics.com/) if you haven’t already, as there are many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples, and frequently asked questions are often addressed there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us better understand and debug the issue.

If this is a custom training or evaluation ❓ Question, please provide as much detail as possible, including relevant code snippets, configuration details, dataset examples, and full logs. Also, check out our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to ensure optimal performance.

Join the Ultralytics community for more support and discussions:
- Real-time chat: [Discord](https://discord.com/invite/ultralytics) 🎧
- In-depth questions: [Discourse](https://community.ultralytics.com/)
- Community threads: [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

Make sure you’re using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for helping improve Ultralytics!

**Comment by glenn-jocher** [comment]
Hi @caisamuels, 

You can limit validation to specific classes using the `classes` parameter with YOLO.val(). This will restrict both prediction and ground truth evaluations to only the classes you specify:

```python
from ultralytics import YOLO

# Load your model
model = YOLO('yolo11n.pt')  # or your custom model path

# Run validation on specific classes only (e.g., classes 0 and 1)
metrics = model.val(data="your_data.yaml", classes=[0, 1])
```

The same works with the CLI:
```
yolo val model=path/to/model.pt data=your_data.yaml classes=0,1
```

This will ensure that only predictions and ground truth for your specified classes are included in the evaluation metrics, which should resolve your false positive issue. The validation process will ignore both predictions and ground truth for any classes not in your specified list.

**Comment by caisamuels** [comment]
> Hi [@caisamuels](https://github.com/caisamuels),
> 
> You can limit validation to specific classes using the `classes` parameter with YOLO.val(). This will restrict both prediction and ground truth evaluations to only the classes you specify:
> 
> from ultralytics import YOLO
> 
> # Load your model
> model = YOLO('yolo11n.pt')  # or your custom model path
> 
> # Run validation on specific classes only (e.g., classes 0 and 1)
> metrics = model.val(data="your_data.yaml", classes=[0, 1])
> The same works with the CLI:
> 
> ```
> yolo val model=path/to/model.pt data=your_data.yaml classes=0,1
> ```
> 
> This will ensure that only predictions and ground truth for your specified classes are included in the evaluation metrics, which should resolve your false positive issue. The validation process will ignore both predictions and ground truth for any classes not in your specified list.

I thought that wasn’t a valid parameter?

**Comment by Y-T-G** [comment]
It's a valid argument.

https://github.com/ultralytics/ultralytics/issues/20248#issuecomment-2817395938

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged in the linked PR: https://github.com/ultralytics/ultralytics/pull/20250. This update introduces a new classes argument to the validation documentation, allowing users to specify which class IDs they want to evaluate during validation. This addition makes it possible to focus validation on selected object classes, enabling more targeted and efficient evaluation—especially useful when only certain classes are relevant to a project.

Special thanks to @Y-T-G for authoring the PR and @Laughing-q for their valuable contributions.

To test if these changes resolve your issue, try installing the latest version from the main branch with:

pip install git+https://github.com/ultralytics/ultralytics.git@main

Alternatively, you can wait for the next official PyPI release, where this update will be included.

Feedback on whether this resolves your issue would be greatly appreciated! Thank you for reporting the issue, and please let us know if you have any further questions or if the issue persists.
